hvplot/util.py

```diff
@@ -237,18 +237,21 @@ def is_geopandas(data):
 
 def process_xarray(data, x, y, by, groupby, use_dask, persist, gridded, label, value_label):
     import xarray as xr
-    dataset = data
-    data_vars = list(dataset.data_vars) if isinstance(data, xr.Dataset) else [data.name]
+    if isinstance(data, xr.Dataset):
+        dataset = data
+    else:
+        name = data.name or label or value_label
+        dataset = data.to_dataset(name=name)
+
+    data_vars = list(dataset.data_vars)
     ignore = (by or []) + (groupby or [])
-    dims = [c for c in data.coords if data[c].shape != () and c not in ignore][::-1]
-    index_dims = [d for d in dims if d in data.indexes]
+    dims = [c for c in dataset.coords if dataset[c].shape != () and c not in ignore][::-1]
+    index_dims = [d for d in dims if d in dataset.indexes]
 
     if gridded:
         data = dataset
-        if data_vars == [None]:
-            label = label or value_label
-            data = data.to_dataset(name=label)
-            data_vars = [label]
+        if len(dims) < 2:
+            dims += [dim for dim in list(data.dims)[::-1] if dim not in dims]
```

Comment:
Could you explain this?

Reply:
yeah I was trying to take whatever is already in `dims` and extend it with any `data.dims` that aren't included. Also I wanted to flip the order in which I add the `data.dims` so that image works out right for instance.

Reply:
the `dims` are set above from coords.

Reply:
>Also I wanted to flip the order in which I add the data.dims so that image works out right for instance.

This worries me a little bit, but I guess it's not an issue because of the conditional (`if len(dims) < 2:`).

Reply:
Right that is why it doesn't break pre-existing behavior. I know it it strange but I think it is very nice to do the right thing with an image with no coords.

Reply:
Okay, sounds fine. My only question is about the case where ``len(dims) == 1``, I think that should be unchanged. If there are coordinates defined it should use those and if there aren't it should fall back to the dims, but I wouldn't mix the two cases.

Reply:
why not mix? I tested that scenario in the tests and it works fine.

Reply:
It seems to me that if coordinates were declared there was probably a reason for declaring coordinates for certain dimensions but not others, so it seems to me like this is dropping information the user has provided.

Reply:
I might not be thinking about this correctly though and ignoring the dimensions would actually result in an error or something.

Reply:
I think ignoring the dimensions would result in an error. The coordinates will always go in first and if the user has defined x and y then those will be used.

Reply:
Yeah I ran the relevant test on master:

```python
hvplot/tests/testutil.py::TestProcessXarray::test_process_2d_xarray_dataarray_with_coords_as_gridded FAILED [100%]

=================================================== FAILURES ====================================================
___________________ TestProcessXarray.test_process_2d_xarray_dataarray_with_coords_as_gridded ___________________

self = <hvplot.tests.testutil.TestProcessXarray testMethod=test_process_2d_xarray_dataarray_with_coords_as_gridded>

    def test_process_2d_xarray_dataarray_with_coords_as_gridded(self):
        import xarray as xr

        da = xr.DataArray(
            data=np.random.randn(4,5),
            coords={'y': [3, 4, 5, 6, 7]},
            dims=['x', 'y'])

        kwargs = self.default_kwargs
        kwargs.update(gridded=True)

>       data, x, y, by, groupby = process_xarray(data=da, **kwargs)

hvplot/tests/testutil.py:119:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

data = <xarray.Dataset>
Dimensions:  (x: 4, y: 5)
Coordinates:
  * y        (y) int64 3 4 5 6 7
Dimensions without coordinates: x
Data variables:
    value    (x, y) float64 -1.406 1.064 0.0174 1.719 ... 0.1767 0.4939 -1.138
x = None, y = None, by = None, groupby = None, use_dask = False, persist = False, gridded = True, label = 'value'
value_label = 'value'

    def process_xarray(data, x, y, by, groupby, use_dask, persist, gridded, label, value_label):
        import xarray as xr
        dataset = data
        data_vars = list(dataset.data_vars) if isinstance(data, xr.Dataset) else [data.name]
        ignore = (by or []) + (groupby or [])
        dims = [c for c in data.coords if data[c].shape != () and c not in ignore][::-1]
        index_dims = [d for d in dims if d in data.indexes]

        if gridded:
            data = dataset
            if data_vars == [None]:
                label = label or value_label
                data = data.to_dataset(name=label)
                data_vars = [label]
            if not (x or y):
>               x, y = index_dims[:2] if len(index_dims) > 1 else dims[:2]
E               ValueError: not enough values to unpack (expected 2, got 1)

hvplot/util.py:253: ValueError
==================================== 1 failed, 7 deselected in 2.23 seconds =====================================

Reply:
So I think the it is better to plot something and let the user say "hmm, that's not what I meant" than to plot nothing.